### ISSUE TITLE
Spec out Keccak permutation

### ISSUE THREAD

**Issue by ChihChengLiang** [appliedzkp/zkevm-specs#142]
We should add the spec of the keccak circuit here so that it can be audited and reviewed.

Created this issue for the permutation part

**Comment by ChihChengLiang** [comment]
Existing one here https://hackmd.io/sK7v0lr8Txi1bgION1rRpw. Now we learned so much, many can be written in a better way.

**Comment by CPerezz** [comment]
Isn't permutation already done @ChihChengLiang .

Padding is not the focus together with arbitrary-lenght inputs and I belive there's another issue for it.

Closing this maybe?

**Comment by ChihChengLiang** [comment]
Yes, the padding part is here https://github.com/appliedzkp/zkevm-specs/issues/143.
The permutation is implemented but I'm thinking to move the hackmd spec back to this repo. What do you think?

**Comment by CPerezz** [comment]
Sorry @ChihChengLiang I oversaw this.

Whatever you prefer. LGTM to move it. 
We could also add the diagrams made for the ZKEVM day at Devconnect to make the circuit a bit more visual.
